### PR TITLE
Preferences Modal UI

### LINKED ISSUE #11: User Preferences UI
## Summary
UI for asking user preference. Below is the preferences list. Some data need to added such as `interests`.

[Prompt Design](https://docs.google.com/presentation/d/1Fgpg78PUIaWHCCe19t24XhX85p6kxJcB/edit#slide=id.g25acae38db3_2_5)

### PR DESCRIPTION
**Pull request by chanon-mike**
## Background
Add a modal for asking user travel preferences such as date, trip type, pace, budget and interests

## Summary
Implement a modal asking for user preferences after submit the data in the input bar
[Figma on planning section](https://www.figma.com/file/TXTmhow5ovM3eZjOZy3ID5/FirstDraft-HomePage?type=design&node-id=35-616&mode=design&t=m36hKQ6PFOIhSnC4-0)

I made the state management logic into a `usePreferences` hook, and pass it as props for each respective form for separation of concerns and making the code short and readable

image

## Test Cases
- [ ] Enter information in input bar
- [ ] Modal appeared, with date, trip type, pace, budget and interests label
- [ ] Enter information and close modal, open modal again and data is saved
- [ ] Change language to Japanese and English display in each respective language

### CODE REVIEW

**Comment by chanon-mike** [comment]
Oh, I forgot that we have the amount of people too... 
Gonna add it!

**Comment by student158** [comment]
Nice UI! There are 2 things I found so far
1. If user press "Enter" without input, the input pane still appear
image

2. Date box still allow impossible date. I think we can make the second box disabled until a date is inputed in the first box.
image

File: frontend/src/components/prompt/form/BudgetForm.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+};
+
+const PaceForm = ({ selectedBudget, handleSelectBudget }: Props) => {
+  const { t } = createTranslation('home');
+
+  return (
+    <>
+      <Typography variant="h6">{t('budget-label')}</Typography>
+      <Select value={selectedBudget} onChange={handleSelectBudget}>
+        <MenuItem value="安い">{t('budget.low')}</MenuItem>
```
Comment (phattd15) [inline frontend/src/components/prompt/form/BudgetForm.tsx]: I'm unsure if should we hardcode the values here

File: frontend/src/components/prompt/form/TripTypeForm.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+const TripTypeForm = ({ selectedTripTypes, handleSelectTripType }: Props) => {
+  const { t } = createTranslation('home');
+
+  return (
+    <>
+      <Typography variant="h6">{t('trip-type-label')}</Typography>
+      <Select value={selectedTripTypes} onChange={handleSelectTripType}>
+        <MenuItem value="一人">{t('trip-type.solo')}</MenuItem>
```
Comment (phattd15) [inline frontend/src/components/prompt/form/TripTypeForm.tsx]: Same concern as the above

**Review by phattd15 — COMMENTED** [review]
I left a comment regarding the hardcoded values. The rest seems smooth.

**Comment by student158** [comment]
LGTM!

**Review by student158 — APPROVED** [review]
LGTM!

**Pull request merged**